### PR TITLE
Fix a few minor memory leaks in AppleTLS

### PR DESCRIPTION
**Pull request by chamons**
- Major leak will be fixed in https://github.com/mono/mono/pull/5300
- https://bugzilla.xamarin.com/show_bug.cgi?id=56814

### CODE REVIEW

**Comment by spouliot** [comment]
to be backported to d15.4

**Comment by nberardi** [comment]
happy to see this commit.

File: src/Security/SslContext.cs [hunk 1]
```diff
@@ -523,7 +523,7 @@ public SslStatus SetEncryptionCertificate (SecIdentity identify, IEnumerable<Sec
 			get {
 				IntPtr value;
 				result = SSLCopyPeerTrust (Handle, out value);
-				return (value == IntPtr.Zero) ? null : new SecTrust (value);
+				return (value == IntPtr.Zero) ? null : new SecTrust (value, true);
```
Comment (OneMHz) [inline src/Security/SslContext.cs]: Looking at SecTrust, passing true to the owns parameter prevents SecTrust from retaining the handle, but in Dispose, it calls release as long as handle is non-zero. What prevents that from releasing a handle SecTrust didn't retain?

Comment (baulig) [inline src/Security/SslContext.cs]: Look at `SSLCopyPeerTrust()` - you must release the returned object when you're done with it.

Comment (OneMHz) [inline src/Security/SslContext.cs]: I'm not sure if I'm understanding your answer. My question is around a situation like this:

```
var x = new SecTrust(handle, true);

// do some stuff

x.Dispose();

```
May be a misuse of it, but it also may not be obvious that handle is going to get an extra CFRelease inside Dispose.

Comment (chamons) [inline src/Security/SslContext.cs]: The second param to SecTrust (handle, true) is transferring ownership to SecTrust, so it knows to dispose the handle when it's done. The default is false, which is why this was a leak.

Comment (spouliot) [inline src/Security/SslContext.cs]: @OneMHz it's the **Create** (Copy) rule [1] that applies - that's what the reference to `SSLCopyPeerTrust` meant.  The instance we get is already retained but it's our job to release it. 

That differs from the **Get** rule where we are responsible to retain the instance we get (and also to release it). In both cases we need to release the instance, it's only how we initialize that differs.

[1] https://developer.apple.com/library/content/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-103029

**Pull request merged**